### PR TITLE
build(*): switch to npm (refs #133)

### PR DESCRIPTION
**Pull request by Aladdin-ADD**
* require node>=8.0.0, npm>=6.0.0
* disable package-lock, no much need to lock a lib

### CODE REVIEW

**Comment by g-plane** [comment]
Why don't you let npm generate lock file?

**Comment by g-plane** [comment]
But, the lock file is for ourselves, not end users.

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,18 +10,15 @@ jobs:
...
       - run: node --version
       - run: npm --version
-      - run: yarn --version
       - restore_cache:
           keys:
-          - v1-dependencies-{{ checksum "yarn.lock" }}
           - v1-dependencies-
 
-      - run: yarn install --frozen-lockfile
+      - run: npm install
```
Comment (g-plane) [inline .circleci/config.yml]: Run `npm ci` instead.

File: .travis.yml [hunk 2]
```diff
@@ -15,13 +15,13 @@ branches:
...
 
-cache: yarn
+cache: npm
 
 matrix:
   fast_finish: true
 
 install:
-  - yarn install
+  - npm install
```
Comment (g-plane) [inline .travis.yml]: Run `npm ci` instead.

File: .circleci/config.yml (hunk 1 above)
Comment (Aladdin-ADD) [reply .circleci/config.yml]: could you please explain a little -- I'm not familar with `npm ci`.

Comment (g-plane) [reply .circleci/config.yml]: https://docs.npmjs.com/cli/ci
https://medium.com/@tomastrajan/how-to-speed-up-continuous-integration-build-with-new-npm-ci-and-package-lock-json-7647f91751a

Comment (g-plane) [reply .circleci/config.yml]: Additionally, it needs `package-lock.json`.

**Comment by fkleuver** [comment]
`npm ci` is good and all (in fact it's the best thing there is at the moment) but we simply can't make the switch as-is because stuff will break. Need the yarn workspaces for lerna to work properly, and `npm ci` only works on npm 6+ which doesn't play very well with node 6 and below (which it also needs to work on).
For the remarks suggesting to get rid of lockfiles: we most certainly need lockfiles, and under no circumstance will we get rid of lockfiles. It has happened before, and it will happen again: dev deps introduce a breaking change in a minor or patch version, and we're screwed.

**Comment by g-plane** [comment]
@fkleuver According to [this issue](https://github.com/yarnpkg/yarn/issues/5891), should we switch the registry to <https://registry.npmjs.org/>?

**Comment by fkleuver** [comment]
Isn't that something you can configure on a machine level?

**Comment by g-plane** [comment]
Machine level? What does you mean?

**Comment by g-plane** [comment]
You're wrong. Once the lockfile, Yarn will use the registry in the lockfile, instead of the settings on your computer.

**Comment by Aladdin-ADD** [comment]
well, I was also wondering whether it's worthy to make the switching at the moment. But an obvious advantange is npm is shipped with Node.js, no need to install -- not all contributors were using yarn(like me). maybe we can reconsider the issue when someday Node.js 8 hit EOF.

**Comment by fkleuver** [comment]
I'm planning on doing a full matrix-test with docker kubernetes to ensure all the bundled outputs are still installable and working in all browsers and module loaders. Once that's in place, we can safely look for improvements in this area.
I do agree that npm is better than yarn from a contributor's perspective. TypeScript project system is around the corner which should support lerna in a more native and straight-forward way. Then we can make the switch

**Comment by g-plane** [comment]
In fact installing Yarn is very easy, and there is installation guide on https://yarnpkg.com . What we need to do is update the contributing guide and state that contributors need to install Yarn, which is just like many other projects which are using Yarn.

**Pull request closed without merging**